app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/MapsActivity.java

```diff
@@ -49,8 +78,49 @@ protected void onCreate(Bundle savedInstanceState) {
         RecyclerView mRecView = this.findViewById(R.id.recycler_view_maps);
         mController.mRecView = mRecView;
         mController.setDynamicRecyclerView(this);
-        mSearchView = this.findViewById(R.id.tb_toolbarsearch);
-        mController.mSearchView = mSearchView;
+
+        setUpSearch();
+    }
+
+    private void setUpSearch(){
+        mSearchView = (FloatingSearchView) findViewById(R.id.search_view);
+        mSearchView.setOnQueryChangeListener(new FloatingSearchView.OnQueryChangeListener() {
+            @Override
+            public void onSearchTextChanged(String oldQuery, String newQuery) {
+            Map<String, String> map = new HashMap<String, String>();
```

Comment:
It is not a great idea to make a network request on every letter typed. Can you do something like this? https://stackoverflow.com/questions/12142021/how-can-i-do-something-0-5-second-after-text-changed-in-my-edittext

Reply:
Got that working on a delay of 250ms for faster responses, if the server starts getting overwhelmed we can lower the refresh rate.